lib/Doctrine/ODM/PHPCR/Query/Builder/ConverterPhpcr.php

```diff
@@ -332,4 +332,35 @@ protected function walkOperandDynamicField(OperandDynamicField $node)
 
         return $op;
     }
+
+    protected function walkOperandStaticLiteral(OperandStaticLiteral $node)
+    {
+        $value = $node->getValue();
+
+        if ($field = $node->getParent()->getChildOfType(AbstractLeafNode::NT_OPERAND_DYNAMIC)) {
+            if ($field instanceof OperandDynamicField) {
+                $meta = $this->aliasMetadata[$field->getAlias()];
+                $type = $meta->getFieldMapping($field->getField())['type'];
+
+                $typeMapping = array(
+                    'string' => 'string',
+                    'long' => 'integer',
+                    'decimal' => 'string',
```

Comment:
decimal is arbitrary large numbers, and the idea in PHPCR is to use the BCMATH extension for this. and that is based on strings.